components/new_channel_modal/new_channel_modal.tsx

```diff
@@ -116,12 +135,54 @@ const NewChannelModal = () => {
             }
 
             handleOnModalCancel();
+
+            // If template selected, create a new board from this template
+            if (addBoard && selectedBoardTemplate) {
+                addBoardToChannel(newChannel.id);
+            }
+
             dispatch(switchToChannel(newChannel));
         } catch (e) {
             onCreateChannelError({message: formatMessage({id: 'channel_modal.error.generic', defaultMessage: 'Something went wrong. Please try again.'})});
         }
     };
 
+    const addBoardToChannel = async (channelId: string) => {
+        const {data: newBoard} = await dispatch(createBoardFromTemplate(selectedBoardTemplate!.id));
+        if (newBoard) {
+            const boardPatch: BoardPatch = {
+                channelId,
+                deletedCardProperties: [],
+                deletedProperties: [],
+                updatedCardProperties: [],
+                updatedProperties: {},
+            };
+            const {data: patchedBoard} = await dispatch(attachBoardToChannel(newBoard.id, boardPatch));
+
+            if (patchedBoard.channelId === channelId) {
+                const boardUrl = `${getSiteURL()}/boards/team/${currentTeamId}/${newBoard.id}`;
+
+                const newBoardMsg = formatMessage(
+                    {
+                        id: 'channel_modal.board.newBoardCreated2',
+                        defaultMessage: '**{user}** created the board [{board}]({boardUrl}) and linked it to this channel.',
```

Comment:
I don't know how it happens but there's a weird double space in the app coming. Maybe adding a trim to the board title would do the trick?
![image](https://user-images.githubusercontent.com/785518/188694787-d173ad97-b708-417e-a7f8-8a4f58cbbc50.png)

Reply:
ok, will check that. Thanks!